test/test_ea_mu_plus_lambda.py

```diff
@@ -125,3 +128,100 @@ def objective(ind):
+
+
+def test_fitness_has_wrong_type(population_params, genome_params, ea_params):
+    def objective(individual):
+        individual.fitness = int(5.0)  # should raise error since fitness should be float
+        return individual
+
+    pop = cgp.Population(**population_params, genome_params=genome_params)
+    ea = cgp.ea.MuPlusLambda(**ea_params)
+    ea.initialize_fitness_parents(pop, objective)
+
+    with pytest.raises(ValueError):
+        ea.step(pop, objective)
+
+
+def test_initialize_fitness_parents(population_params, genome_params, ea_params):
+    def objective(individual):
+        individual.fitness = -1.0
+        return individual
+
+    pop = cgp.Population(**population_params, genome_params=genome_params)
+
+    ea = cgp.ea.MuPlusLambda(**ea_params)
+    ea.initialize_fitness_parents(pop, objective)
+    assert all([ind.fitness is not None for ind in pop.parents])
+
+
+def test_step(population_params, genome_params, ea_params):
+    def objective(individual):
+        individual.fitness = float(individual.idx)
+        return individual
+
+    pop = cgp.Population(**population_params, genome_params=genome_params)
+
+    ea = cgp.ea.MuPlusLambda(**ea_params)
+    ea.initialize_fitness_parents(pop, objective)
+    old_parent_ids = sorted([ind.idx for ind in pop.parents])
+    ea.step(pop, objective)
+    new_parent_ids = sorted([ind.idx for ind in pop.parents])
+    # After one step, the new parent population should have IDs that
+    # are offset from the old parent ids by n_offsprings
+    # This is by construction in this test because the fitness is equal to the id
+    assert all(
+        [
+            new_id == old_id + ea_params["n_offsprings"]
+            for new_id, old_id in zip(new_parent_ids, old_parent_ids)
+        ]
+    )
+
+
+def test_sort(population_params, genome_params, ea_params):
+    def objective(individual):
+        individual.fitness = float(individual.idx)
+        return individual
+
+    pop = cgp.Population(**population_params, genome_params=genome_params)
+    ea = cgp.ea.MuPlusLambda(**ea_params)
+    ea.initialize_fitness_parents(pop, objective)
+    sorted_parents = ea._sort(pop.parents)
+    # Assert that the sorting inverted the list of parents (because the fitness is equal to the id)
+    assert sorted_parents == pop.parents[::-1]
+
+
+def test_create_new_offspring_and_parent_generation(population_params, genome_params, ea_params):
+    def objective(individual):
+        individual.fitness = float(individual.idx)
+        return individual
+
+    pop = cgp.Population(**population_params, genome_params=genome_params)
+    ea = cgp.ea.MuPlusLambda(**ea_params)
+
+    ea.initialize_fitness_parents(pop, objective)
+
+    offsprings = ea._create_new_offspring_generation(pop)
+    assert len(offsprings) == ea_params["n_offsprings"]
+    assert all([ind.idx >= pop.n_parents for ind in offsprings])
+    # Assert that all offspring dna are different from all parents dna
```

Comment:
whether this is true will depend on our decision in #172. no need to change this test now, just making sure we're on the same page

Reply:
Yes, if we're following my suggestion, with a small chance, there might be no change at all. However, we could control for that in this case here by setting the mutation rate to 1 (or approx. 1).